securedrop/models.py

```diff
@@ -710,15 +710,27 @@ def validate_api_token_and_get_user(token: str) -> 'Union[Journalist, None]':
 
         return Journalist.query.get(data['id'])
 
-    def to_json(self) -> 'Dict[str, Union[str, bool, str]]':
+    def to_json(self, bare: bool = False) -> 'Dict[str, Union[str, bool, str]]':
+        """Returns a JSON representation of the journalist user. If a bare
+           representation is requested, potentially sensitive or extraneous
+           fields are excluded."""
+
+        try:
+            last_login = self.last_access.isoformat() + 'Z'
```

Comment:
When I first attempted to request a `to_json` representation of all users in the dev env, this threw an exception due to this attribute never having been set. While not required for the bare representation, it seemed sensible to add a check here.

Reply:
i think it makes sense to also add an "all_details" check that defaults to false for the "/user" endpoint. otherwise, we'll have some admins with `is_admin` set and some admins without `is_admin` set depending on who has logged into the workstation. looking at the table directly as an admin could be misleading. also, we don't do anything with this information in the client and should probably wait to include it once we finish our client admin stories.

Reply:
>  you could set last_login within the bare clause where `bare` is false since that's the only place it's used

That makes sense; we'll still have to do the `try` check to avoid errors on newly created users, but I'll move it.

> instead of "bare" you could call it "all_details" or "all_info"

I like `all_info`, will use that. I was trying to avoid suggesting that it's a _full_ representation, since both representations exclude credentials.

Reply:
> also add an "all_details" check that defaults to false for the "/user" endpoint

Could you clarify? This PR doesn't modify the behavior of the `/user` endpoint (it currently already includes the `is_admin` Boolean), do you think that it should?

Reply:
Made the other two changes in aa0a5914e5dc55e6c936e5bd2470d37f32f83122

Reply:
> Could you clarify? This PR doesn't modify the behavior of the /user endpoint (it currently already includes the is_admin Boolean), do you think that it should?

Oh, I see, I forgot that we don't store the `is_admin` result from the `/user` endpoint so we don't have to worry about the situation I brought up where some admin acounts will have `is_admin` set and some will not. Okay, ignore previous comment.